Index: util/src/main/kotlin/ie/dublinmapper/util/Operator.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package ie.dublinmapper.util\n\nenum class Operator {\n\n    DART,\n    DUBLIN_BIKES,\n    DUBLIN_BUS,\n    LUAS\n\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- util/src/main/kotlin/ie/dublinmapper/util/Operator.kt	(revision 72ca16c211c310ad126828adc5f93b871f9b74ca)
+++ util/src/main/kotlin/ie/dublinmapper/util/Operator.kt	(date 1546908113113)
@@ -1,10 +1,38 @@
 package ie.dublinmapper.util
 
-enum class Operator {
+import java.util.*
+
+enum class Operator(
+    val fullName: String,
+    val shortName: String
+) {
+
+    COMMUTER("Commuter", "COMM"),
+    DART("DART", "DART"),
+    DUBLIN_BIKES("Dublin Bikes", "BIKE"),
+    DUBLIN_BUS("Dublin Bus", "BAC"),
+    GO_AHEAD("Go Ahead", "GAD"),
+    INTERCITY("InterCity", "ICTY"),
+    LUAS("Luas", "LUAS");
+
+    companion object {
 
-    DART,
-    DUBLIN_BIKES,
-    DUBLIN_BUS,
-    LUAS
+        fun allRail(): EnumSet<Operator> {
+            return EnumSet.of(COMMUTER, DART, INTERCITY)
+        }
+
+        fun dublinBusAndGoAhead(): EnumSet<Operator> {
+            return EnumSet.of(DUBLIN_BUS, GO_AHEAD)
+        }
+
+        fun dublinBusOnly(): EnumSet<Operator> {
+            return EnumSet.of(DUBLIN_BUS)
+        }
+
+        fun goAheadOnly(): EnumSet<Operator> {
+            return EnumSet.of(GO_AHEAD)
+        }
+
+    }
 
 }
Index: domain/src/main/kotlin/ie/dublinmapper/domain/model/ServiceLocation.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package ie.dublinmapper.domain.model\n\nimport ie.dublinmapper.util.Coordinate\nimport ie.dublinmapper.util.Operator\n\ninterface ServiceLocation {\n\n    val id: String\n\n    val name: String\n\n    val coordinate: Coordinate\n\n    val operator: Operator\n\n    val mapIconText: String\n\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- domain/src/main/kotlin/ie/dublinmapper/domain/model/ServiceLocation.kt	(revision 72ca16c211c310ad126828adc5f93b871f9b74ca)
+++ domain/src/main/kotlin/ie/dublinmapper/domain/model/ServiceLocation.kt	(date 1546908113105)
@@ -2,6 +2,7 @@
 
 import ie.dublinmapper.util.Coordinate
 import ie.dublinmapper.util.Operator
+import java.util.*
 
 interface ServiceLocation {
 
@@ -11,7 +12,9 @@
 
     val coordinate: Coordinate
 
-    val operator: Operator
+    val operators: EnumSet<Operator>
+
+    val routes: Map<Operator, Set<String>>
 
     val mapIconText: String
 
